storage/object_storage.go

```diff
@@ -0,0 +1,179 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"net/url"
+	"path"
+	"strings"
+)
+
+type ObjectStorageConfig struct {
+	Provider        string
+	Endpoint        string
+	Region          string
+	AccessKeyId     string
+	SecretAccessKey string
+}
+
+const (
+	DefaultRegion = "cn-northwest-1"
```

Comment:
Maybe we should not use a default region.

Reply:
It's useless but used in s3 compatible object storage. `DummyRegion` is better.